File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.go
func solution(n int, m int, section []int) int {
    count := 0
    paintedUntil := 0

    for _, s := range section {
        if s > paintedUntil {
            paintedUntil = s + m - 1
            count++
        }
    }

    return count
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    
    answer = 0
    start = section[0]
    
    for i in range(1, len(section)):
        
        if section[i] - start >= m:
            answer += 1
            start = section[i]
        
    
    return answer + 1